mmaction/models/backbones/resnet_tin.py

```diff
@@ -0,0 +1,370 @@
+    # offset0, offset1: [N, num_segments]
+    offset0 = torch.floor(offset).int()
+    offset1 = offset0 + 1
+
+    # data, data0, data1: [N, num_segments, C, H * W]
+    data = data.view(n, t, c, h * w).contiguous()
+    data0 = tin_shift(data, offset0)
+    data1 = tin_shift(data, offset1)
+
+    # weight0, weight1: [N, num_segments]
+    weight0 = 1 - (offset - offset0.float())
+    weight1 = 1 - weight0
+
+    # weight0, weight1:
+    # [N, num_segments] -> [N, num_segments, C // num_segments] -> [N, C]
+    group_size = offset.shape[1]
+    weight0 = weight0[:, :, None].repeat(1, 1, c // group_size)
+    weight0 = weight0.view(weight0.size(0), -1)
+    weight1 = weight1[:, :, None].repeat(1, 1, c // group_size)
+    weight1 = weight1.view(weight1.size(0), -1)
+
+    # weight0, weight1: [N, C] -> [N, 1, C, 1]
+    weight0 = weight0[:, None, :, None]
+    weight1 = weight1[:, None, :, None]
+
+    # output: [N, num_segments, C, H * W] -> [N, num_segments, C, H, W]
+    output = weight0 * data0 + weight1 * data1
+    output = output.view(n, t, c, h, w)
+
+    return output
+
+
+class CombineNet(nn.Module):
+    """Combine Net.
+
+    It combines Temporal interlace module with some part of ResNet layer.
+
+    Args:
+        net1 (nn.module): Temporal interlace module.
+        net2 (nn.module): Some part of ResNet layer.
+    """
+
+    def __init__(self, net1, net2):
+        super().__init__()
+        self.net1 = net1
+        self.net2 = net2
+
+    def forward(self, x):
+        """Defines the computation performed at every call.
+
+        Args:
+            x (torch.Tensor): The input data.
+
+        Returns:
+            torch.Tensor: The output of the module.
+        """
+        # input shape: [num_batches * num_segments, C, H, W]
+        # output x shape: [num_batches * num_segments, C, H, W]
+        x = self.net1(x)
+        # [num_batches * num_segments, C, H, W]
+        x = self.net2(x)
+        return x
+
+
+class WeightNet(nn.Module):
+    """WeightNet in Temporal interlace module.
+
+    The WeightNet consists of two parts: one convolution layer
+    and a sigmoid function. Following the convolution layer, the sigmoid
+    function and rescale module can scale our output to the range (0, 2).
+    Here we set the initial bias of the convolution layer to 0, and the
+    final initial output will be 1.0.
+
+    Args:
+        in_channels (int): Channel num of input features.
+        groups (int): Number of groups for fc layer outputs.
+    """
+
+    def __init__(self, in_channels, groups):
+        super().__init__()
+        self.sigmoid = nn.Sigmoid()
+        self.groups = groups
+
+        self.conv = nn.Conv1d(in_channels, groups, 3, padding=1)
+
+        self.init_weights()
+
+    def init_weights(self):
+        """Initiate the parameters either from existing checkpoint or from
+        scratch."""
+        # we set the initial bias of the convolution
+        # layer to 0, and the final initial output will be 1.0
```

Comment:
what is "final initial" output?

Reply:
![image](https://user-images.githubusercontent.com/33949370/91289404-e2a08b80-e7c4-11ea-92b2-b9f32ec24329.png)
This comment is the same with the original repo and the paper. It means the initialized conv layer will output 1.0

Reply:
what's the initial weights?

Reply:
The conv weights are initialized by default using pytorch `kaiming_uniform_` or loading from ckpt, the codes only changes the bias as the paper said.